runtime/drivers/druid/druidsqldriver/druid_api_sql_driver.go

```diff
@@ -0,0 +1,267 @@
+var _ driver.QueryerContext = &sqlConnection{}
+
+func emptyTransformer(v any) (any, error) {
+	return v, nil
+}
+
+func toStringArray(values []any) []string {
+	s := make([]string, len(values))
+	for i, v := range values {
+		vv, ok := v.(string)
+		if !ok {
+			return nil
+		}
+		s[i] = vv
+	}
+	return s
+}
+
+func (c *sqlConnection) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Rows, error) {
+	b, err := json.Marshal(druidRequest(query, args))
+	if err != nil {
+		return nil, err
+	}
+
+	bodyReader := bytes.NewReader(b)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.dsn, bodyReader)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	dec := json.NewDecoder(resp.Body)
+
+	var obj any
+	err = dec.Decode(&obj)
+	if err != nil {
+		resp.Body.Close()
+		return nil, err
+	}
+	switch v := obj.(type) {
+	case map[string]any:
+		resp.Body.Close()
+		return nil, fmt.Errorf("%v", obj)
+	case []any:
+		columns := toStringArray(v)
+		err = dec.Decode(&obj)
+		if err != nil {
+			resp.Body.Close()
+			return nil, err
+		}
+
+		types := toStringArray(obj.([]any))
+
+		transformers := make([]func(any) (any, error), len(columns))
+		for i, c := range types {
+			transformers[i] = emptyTransformer
+			if c == "TIMESTAMP" {
+				transformers[i] = func(v any) (any, error) {
+					t, err := time.Parse(time.RFC3339, v.(string))
+					if err != nil {
+						return nil, err
+					}
+
+					return t, nil
+				}
+			} else if c == "ARRAY" {
+				transformers[i] = func(v any) (any, error) {
+					var l []any
+					err := json.Unmarshal([]byte(v.(string)), &l)
+					if err != nil {
+						return nil, err
+					}
+					return l, nil
+				}
+			} else if c == "OTHER" {
+				transformers[i] = func(v any) (any, error) {
+					var l map[string]any
+					err := json.Unmarshal([]byte(v.(string)), &l)
+					if err != nil {
+						return nil, err
+					}
+					return l, nil
+				}
+			}
+		}
+
+		druidRows := &druidRows{
+			closer:       resp.Body,
+			dec:          dec,
+			columns:      columns,
+			types:        types,
+			transformers: transformers,
+		}
+		return druidRows, nil
+	default:
+		resp.Body.Close()
+		return nil, fmt.Errorf("unexpected response: %v", obj)
+	}
+}
+
+func toType(v any) string {
+	switch v.(type) {
+	case int:
+		return "INTEGER"
+	case float64:
+		return "DOUBLE"
+	case bool:
+		return "BOOLEAN"
+	default:
+		return "VARCHAR"
+	}
+}
+
+type druidRows struct {
+	closer       io.ReadCloser
+	dec          *json.Decoder
+	columns      []string
+	types        []string
+	transformers []func(any) (any, error)
+}
+
+var _ driver.Rows = &druidRows{}
+
+func (dr *druidRows) Columns() []string {
+	return dr.columns
+}
+
+func (dr *druidRows) Close() error {
+	return dr.closer.Close()
+}
+
+func (dr *druidRows) Next(dest []driver.Value) error {
+	var a []any
+	err := dr.dec.Decode(&a)
```

Comment:
Instead of allocating a new array on each call to `Next`, could we maybe move it to a property on the `druidRows` and re-use the same array on each iteration?